src/modules/components/proposal/proposalActions/proposalActionsActionVerfication/proposalActionsActionVerfication.tsx

```diff
@@ -0,0 +1,56 @@
+import { type HTMLAttributes } from 'react';
+import { Heading, Icon, IconType } from '../../../../../core';
+import { addressUtils } from '../../../../utils';
+import type { IProposalAction, IProposalActionWithdrawToken } from '../proposalActionsTypes';
+import { ProposalActionType } from '../proposalActionsTypes/proposalAction';
+
+export interface IProposalActionsActionVerificationProps extends HTMLAttributes<HTMLDivElement> {
+    /**
+     * Proposal action base
+     */
+    action: IProposalAction;
+}
+
+export const ProposalActionsActionVerification: React.FC<IProposalActionsActionVerificationProps> = (props) => {
+    const { action } = props;
+
+    let verificationLabel: string | undefined;
+    let iconType = IconType.SUCCESS;
+    let contractAddressClassName: string | undefined;
+    let iconClassName: string | undefined;
+
+    if (action.inputData == null) {
+        iconType = IconType.WARNING;
+        contractAddressClassName = 'text-warning-800';
+        iconClassName = 'text-warning-500';
+    } else {
+        contractAddressClassName = 'text-neutral-500';
+        iconClassName = 'text-primary-300';
+        switch (action.type) {
+            case ProposalActionType.WITHDRAW_TOKEN: {
+                const withdrawAction = action as IProposalActionWithdrawToken;
+                verificationLabel = withdrawAction.token.name;
+                break;
+            }
+
+            default: {
+                verificationLabel = 'Verified';
+                break;
+            }
+        }
+    }
```

Comment:
This can be simplified by using a ternary operator as we don't need the switch statement:
```
const contractClassName = action.inputData == null ? 'text-warning-800' : 'text-neutral-500':
const contractName = action.inputData?.contract;
const icon = action.inputData == null
  ? { className: 'text-warning-500', icon: IconType.WARNING }
  : { className: 'text-primary-300', icon: IconType.SUCCESS }
```
The `verificationLabel` should just be the `inputData.contract` which is the contract name the action is interacting with